snapcraft/internal/lxd/_containerbuild.py

```diff
@@ -289,6 +293,7 @@ def _inject_snap(self, name: str, tmp_dir: str):
             ], tmp_dir)
 
         container_filename = os.path.join(os.sep, 'run', filename)
+        os.chmod(filepath, 0o777)
```

Comment:
Is this actually what you want? Or do you simply want to copy the owner's permissions over to everything else? I'd rather be smart about this instead of just setting everything to 777. If you agree, perhaps we should extract [_replicate_owner_mode](https://github.com/snapcore/snapcraft/blob/master/snapcraft/plugins/_python/_pip.py#L61) to be more generally useful.

Reply:
I did some more testing on this. It looks like the `chmod` is not necessary afterall, I'm guessing something else was wrong before and I fixed it in the meanwhile.